Capitalize Valgrind in docs (#41986)
nodejs/node: doc/contributing/investigating-native-memory-leaks.md

# Investigating memory leaks with Valgrind

>  A Node.js process may run out of memory due to excessive consumption of
>  native memory. Native Memory is memory which is not managed by the
>  V8 Garbage collector and is allocated either by the Node.js runtime, its
>  dependencies or native [addons](https://nodejs.org/docs/latest/api/n-api.html).

Reviewer: ```suggestion
A Node.js process may run out of memory due to excessive consumption of
native memory. Native Memory is memory which is not managed by the
V8 Garbage collector and is allocated either by the Node.js runtime, its
dependencies or native [addons](https://nodejs.org/docs/latest/api/n-api.html).
```

>  This guide provides information on how to use Valgrind to investigate these
>  issues on Linux platforms.

Reviewer: ```suggestion
This guide provides information on how to use Valgrind to investigate these
issues on Linux platforms.
```

>  ## Valgrind

Reviewer: ```suggestion
## Valgrind
```

 [Valgrind](https://valgrind.org/docs/manual/quick-start.html) is a
 tool available on Linux distributions which can be used to investigate
 memory usage including identifying memory leaks (memory which is
 allocated and not freed) and other memory related problems

>  like double freeing memory.

Reviewer: ```suggestion
[Valgrind](https://valgrind.org/docs/manual/quick-start.html) is a
tool available on Linux distributions which can be used to investigate
memory usage including identifying memory leaks (memory which is
allocated and not freed) and other memory related problems
like double freeing memory.
```

>  [Valgrind](https://valgrind.org/docs/manual/quick-start.html) is a
>  tool available on Linux distributions which can be used to investigate
>  memory usage including identifying memory leaks (memory which is
>  allocated and not freed) and other memory related problems
>  like double freeing memory.

Reviewer: ```suggestion
[Valgrind](https://valgrind.org/docs/manual/quick-start.html) is a
tool available on Linux distributions which can be used to investigate
memory usage including identifying memory leaks (memory which is
allocated and not freed) and other memory related problems
like double freeing memory.
```

>  To use Valgrind:

Reviewer: ```suggestion
To use Valgrind:
```

>  * Be patient, running under Valgrind slows execution significantly

Reviewer: ```suggestion
* Be patient, running under Valgrind slows execution significantly
```

>    due to the checks being performed.
>  * Reduce your test case to the smallest reproduce. Due to the slowdown it is

Reviewer: ```suggestion
  due to the checks being performed.
* Reduce your test case to the smallest reproduce. Due to the slowdown it is
```

>    important to run the minimum test case in order to be able to do it in
>    a reasonable time.

Reviewer: ```suggestion
  important to run the minimum test case in order to be able to do it in
  a reasonable time.
```

Now we can see the specific file name and line in the Node.js code which
caused the allocation (inspector\_agent.cc:140).

> Thusly, we can examine the line (and its surrounding code) in order

Reviewer 2: ```suggestion
Thus, we can examine the line (and its surrounding code) in order
```

> Thus, we can examine the line (and its surrounding code) in order
> to find a potential solution for the memory leak.

Reviewer: ```suggestion
We can examine that line (and its surrounding code) to
find a solution for the memory leak.
```
